packages/js/src/ui/api/hooks/useNotifications.ts

```diff
@@ -43,5 +44,10 @@ export const useNotificationsInfiniteScroll = (props: UseNotificationsInfiniteSc
     filter = newFilter;
   });
 
-  return { data, initialLoading, setEl, end };
+  const refetch = async ({ filter }: { filter?: NotificationFilter }) => {
```

Comment:
made it a promise so that we can await for the successful fetch of the notification